src/platforms/macos/daemon/macosdnsmanager.h

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef MACOSDNSMANAGER_H
+#define MACOSDNSMANAGER_H
+
+#include <systemconfiguration/scdynamicstore.h>
+
+#include <QMap>
+
+#include "command.h"
+
+class MacOSDnsManager final : public Command {
+ public:
+  explicit MacOSDnsManager(QObject* parent);
+  ~MacOSDnsManager();
+
+  int run(QStringList& tokens) override;
+
+ private:
+  QStringList enumerateNetServices();
+  static QString scErrorMessage();
+  void restoreSnapshot();
+  void clearSnapshot();
+  int waitForTermination();
+
+ private:
+  SCDynamicStoreRef m_scStore = nullptr;
```

Comment:
Would it be possible to test the functionality of this class by mocking the value of this property? IIUC that is main outside dep?